.github/workflows/etcd-defrag.yaml

```diff
@@ -0,0 +1,48 @@
+name: Publish
+
+on:
+- push
+- workflow_dispatch
+
+jobs:
+
+  publish-tagged:
+    if: |
+      startsWith(github.ref, 'refs/tags/v') &&
+      !github.event.pull_request.head.repo.fork
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.19"
+      - uses: ko-build/setup-ko@v0.6
+        name: Setup ko
+      - name: Run ko publish for ghcr.io
+        env:
+          KO_DOCKER_REPO: ghcr.io/${{ github.repository_owner }}
+        run: |
+          export COMMIT=${{ github.sha}}
+          export TAG=$(echo ${{ github.ref }} | cut -d "/" -f 3 - )
+          ko publish ./ --base-import-paths --platform=linux/amd64,linux/arm64 --tags $TAG
+          ko publish ./ --base-import-paths --platform=linux/amd64,linux/arm64
+
+  publish-untagged:
```

Comment:
What's the motivation to release untagged image?

Reply:
when developping a new feature for example, I like the ability to test it directly through using the `etcd-defrag:dev` tag. could also be named `:nightly`.

this can be removed if you don't see the usage for this

Reply:
I suggest to remove it for now. I do not think there will be frequent changes or features for this repository.

Reply:
good point. done